### PR TITLE
docs(plan): `dedented_multiline_string`

### PR DESCRIPTION
**Pull request by KSXGitHub**
## What

Adds a planning document for a new lint, **`dedented_multiline_string`** (`planned-rules/dedented-multiline-string.md`).

The rule flags string literals broken across source lines by *raw* line breaks, which push the body flush to the left margin and out of the surrounding indentation:

```rust
let justfile = "\
clean:
    rm -rf ./dist
";
```

The diagnostic attaches every applicable alternative fix: for a multi-line value, `include_str!` of a sibling fixture file, a re-indented `text_block_fnl! { ... }` block, and a `concat!`; for a value with a single non-empty line, a one-line `"…\n"`.

## Notes

- It is the source-layout complement of `escaped_multiline_string` (which targets the opposite spelling: `\n` escapes crammed onto one source line). The rules stay **independent** — none defers to or suppresses another. A dedented JSON literal trips both this rule and `manual_json_string`, by design; they flag different defects (source layout vs. hand-rolled JSON construction).
- The pattern already exists in this repo's own test suite; the doc cites the real occurrences (`tests/import_grouping_mismatch_submodules.rs`, `tests/uncombined_self_import_submodules.rs`) and contrasts them with an *indented* raw-string boundary case that does not fire.

## Changed files

- `planned-rules/dedented-multiline-string.md` — new planning doc.
- `planned-rules/README.md` — index entry.
- `planned-rules/escaped-multiline-string.md`, `planned-rules/manual-json-string.md` — reciprocal "independent, no cross-rule avoidance" interaction notes.

Docs only; no code or behavior change.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: planned-rules/manual-json-string.md [hunk 1]
```diff
@@ -534,3 +534,9 @@ entirely.
   should defer to this rule, since the `json!` rewrite
   eliminates the `format!` invocation entirely and the wrap
   question becomes moot.
+- [`deindented-multiline-string`](./deindented-multiline-string.md) —
+  same direction: when a de-indented multi-line literal's content
+  parses as a structurally interesting JSON document,
+  `deindented_multiline_string` emits nothing and lets this rule
+  suggest the `json!` form, which removes the literal — and its
+  de-indentation — entirely.
```
Comment (KSXGitHub) [inline planned-rules/manual-json-string.md]: `deindented_multiline_string` should still trigger on deindented JSON strings.

The two rules do different things. So they are not double-firing.

We need to keep the implementation simple. Avoiding each other is not simple.

File: planned-rules/escaped-multiline-string.md [hunk 2]
```diff
@@ -371,3 +371,17 @@ text_block! {
...
+*source spelling*. This rule targets newlines compressed onto one
+(or few) source lines as `\n` escapes; its sibling
+[`deindented-multiline-string`](./deindented-multiline-string.md)
+targets newlines spelled as *raw* source breaks that drop the
+literal's body out of the surrounding indentation. A literal can
+satisfy both decoded-value triggers, so to avoid double-firing
+`escaped_multiline_string` defers on any literal whose source
+contains a de-indenting raw line break — `deindented_multiline_string`
+owns that case and offers the richer remedy menu (`include_str!`
+extraction, single-line collapse, `json!` deferral).
```
Comment (KSXGitHub) [inline planned-rules/escaped-multiline-string.md]: Hmm. It's not a double-firing.

The two rules do different things.

Neither rules need to be avoid each other. This is to keep the implementation simple.

**Review by KSXGitHub — COMMENTED** [review]
Some suggestions.

File: planned-rules/dedented-multiline-string.md [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+`MachineApplicable` requires *both* that exactly one suggestion is
+enabled in config *and* that the fix is intrinsically exact — only
+`concat!` and the single-line collapse qualify (and only when sole).
+Otherwise every suggestion is `MaybeIncorrect`, so `cargo dylint
+--fix` never silently rewrites to one of several alternatives.
+
+- See [`IMPLEMENTATION_CONVENTIONS.md`](./IMPLEMENTATION_CONVENTIONS.md)
+  for cross-cutting conventions that apply to every rule in this
+  catalogue, in particular the lint-name namespacing
+  (`perfectionist::*`) that every registered lint follows.
```
Comment (KSXGitHub) [inline planned-rules/dedented-multiline-string.md]: Shouldn't this section and the sentence before belong to other sections?

They don't seem to belong in "Difficulty". Am I wrong?

Comment (KSXGitHub) [reply planned-rules/dedented-multiline-string.md]: You're right — neither belonged in "Difficulty". Fixed in 015c119:

- The `MachineApplicable` policy paragraph was a duplicate; the applicability rule already lives in "What to lint" (step 2), so I dropped it rather than re-home it.
- The cross-cutting conventions pointer moved up into "Implementation notes".

"Difficulty" now keeps only the difficulty assessment and the per-fix synthesis bullets that justify the "Medium" rating.

---
_Generated by [Claude Code](https://claude.ai/code)_
